Sources/StreamChatUI/Utils/ImageCDN.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public var cachingKeyProvider: CachingKeyProvider = StreamCachingKeyProvider()
+
+/// Caching key provider is providing custimised (filtered) key to be used for caching.
+///
+/// The URL can contain extra parameters that are changing every time. Those parameters are effectively blocking the caching.
+/// By providing customised (filtered) key, the cache will work as expected.
+public protocol CachingKeyProvider {
+    /// Customised (filtered) key for image cache.
+    /// - Parameter imageURL: URL of the image that should be customised (filtered).
+    /// - Returns: String to be used as an image cache key.
+    func cachingKey(forImage url: URL) -> String
+}
+
+open class StreamCachingKeyProvider: CachingKeyProvider {
+    public static var streamCDNURL = "stream-io-cdn.com"
+    
+    public func cachingKey(forImage url: URL) -> String {
+        let key = url.absoluteString
+        
+        guard
+            var components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let host = components.host
+        else { return key }
+        
+        // Stream CDN
+        if host.contains(StreamCachingKeyProvider.streamCDNURL) {
+            components.query = nil
+            return components.string ?? key
+        }
+        
+        return key
+    }
+}
```

Comment:
This probably doesn't have to be `open` as it's more an implementation detail of the default CDN.